ui/ToggleGroup.tsx

```diff
@@ -0,0 +1,77 @@
+import { Dispatch, ReactNode, SetStateAction } from "react";
+import { RadioGroup, RadioOptionProps } from "@headlessui/react";
+import { cva } from "class-variance-authority";
+
+export const toggleOptionStyles = cva(
+  [
+    "flex items-center py-2 font-medium leading-5 rounded-12",
+    "outline-none cursor-pointer",
+  ],
+  {
+    variants: {
+      active: {
+        true: ["bg-surface-surface-0 shadow-2"],
+        false: ["hover:bg-surface-surface-1 hover:shadow-1"],
+      },
+      size: {
+        lg: ["py-3 px-4 text-md"],
+        md: ["py-2 px-3 text-base"],
+        sm: ["py-2 px-2 text-sm"],
+        xs: ["py-1 px-2 text-2xs"],
+      },
+    },
+    defaultVariants: {
+      size: "sm",
+    },
+  }
+);
+
+export type ToogleOptionSizeProp = "xs" | "sm" | "md" | "lg";
+
+interface ToggleGroupProp {
+  value: string;
+  onChange: Dispatch<SetStateAction<string>>;
+  children: React.ReactNode;
+}
+
+export const ToggleGroup = ({
+  value,
+  onChange,
+  children,
+  ...props
+}: ToggleGroupProp) => {
+  return (
+    <RadioGroup
+      value={value}
+      onChange={onChange}
+      className="flex space-x-1 bg-surface-surface-2 p-[4px] rounded-12 w-fit"
+      {...props}
+    >
+      {children}
+    </RadioGroup>
+  );
+};
+
+export const ToogleGroupLabel = RadioGroup.Label;
+
+interface ToggleOptionProp {
+  children: ReactNode;
+  size?: ToogleOptionSizeProp;
+  value: string;
+}
+export const ToogleGroupOption = ({
+  children,
+  value,
+  size,
+  ...props
+}: ToggleOptionProp) => {
```

Comment:
```suggestion
type ToggleOptionProp = RadioOptionProps<React.ElementType, string> &
  React.PropsWithChildren<{
    size?: ToogleOptionSizeProp;
  }>;

export const ToogleGroupOption = ({
  children,
  size,
  ...props
}: ToggleOptionProp) => {
```

We can use this types and props here. Value can be passed inside `props`